tests/test_cli.py

```diff
@@ -0,0 +1,73 @@
+from unittest.mock import MagicMock
+
+import pytest
+from pytest_mock import MockerFixture
+
+from wanikani_notifier.cli import process_all, processor, generator
+
+
+@pytest.fixture
+def mocked_wk_client(mocker: MockerFixture) -> MagicMock:
+    return mocker.patch("wanikani_notifier.cli.WaniKaniClient")
+
+
+@pytest.fixture
+def mocked_processor_counter(mocker: MockerFixture):
+    return mocker.Mock()
+
+
+@pytest.fixture
+def mocked_consumer(mocked_processor_counter):
+    @processor
+    def message_consumer(_, message_stream):
+        mocked_processor_counter()
+        yield from message_stream
```

Comment:
It should not yield back objects or it should at least acknowledge it in its name

Reply:
```suggestion
```